mkdocs/commands/serve.py

```diff
@@ -1,21 +1,49 @@
 from __future__ import unicode_literals
+
 import logging
 import shutil
 import tempfile
 
+from os.path import isfile, join
 from mkdocs.commands.build import build
 from mkdocs.config import load_config
 
 log = logging.getLogger(__name__)
 
 
+def _get_handler(site_dir):
+
+    from tornado.web import StaticFileHandler
+    from tornado.template import Loader
+
+    class WebHandler(StaticFileHandler):
+
+        def write_error(self, status_code, **kwargs):
+
+            if status_code in (404, 500):
+                error_page = '{}.html'.format(status_code)
+                if isfile(join(site_dir, error_page)):
+                    self.write(Loader(site_dir).load(error_page).generate())
+                else:
+                    super(WebHandler, self).write_error(status_code, **kwargs)
+
+    return WebHandler
+
+
 def _livereload(host, port, config, builder, site_dir):
 
     # We are importing here for anyone that has issues with livereload. Even if
     # this fails, the --no-livereload alternative should still work.
     from livereload import Server
 
-    server = Server()
+    class LiveReloadServer(Server):
+
+        def get_web_handlers(self, script):
+            handlers = super(LiveReloadServer, self).get_web_handlers(script)
+            handlers = handlers.pop()
+            return [(handlers[0], _get_handler(site_dir), handlers[2],)]
```

Comment:
This reads peculiarly. You assign the last item (via pop) in a list of handlers to the `handlers` variable, then you reference items in that last item via index. So `handlers` starts out as a list of lists? 

I'm not questioning that this is correct. But months/years later when someone comes along to work on this code, it may not be immediately obvious whats going on here. Maybe we should use two separate variables (rather than `handlers` for both), or include a comment explaining what is contained in `handlers` on line 42 and how that differs from the new `handlers` on line 43.